v2/fthealth.go

```diff
@@ -0,0 +1,108 @@
+package v2
+
+import (
+	"sync"
+	"time"
+)
+
+type HealthCheck struct {
+	SystemCode  string
+	Name        string
+	Description string
+	Checks      []Check
+	Parallel    bool
+}
+
+type Check struct {
+	ID               string
+	Name             string
+	Severity         uint8
+	BusinessImpact   string
+	TechnicalSummary string
+	PanicGuide       string
+	Checker          func() (string, error)
+}
+
+func RunCheck(hc *HealthCheck) HealthResult {
+	return hc.health()
+}
+
+func (ch *HealthCheck) health() (result HealthResult) {
+	ch.setResultBasicInfo(&result)
+	ch.doChecks(ch.Parallel, &result)
+	setResultGlobalOK(&result)
+	return
+}
+
+func (ch *HealthCheck) setResultBasicInfo(result *HealthResult) {
+	result.SchemaVersion = 1
+	result.SystemCode = ch.SystemCode
+	result.Name = ch.Name
+	result.Description = ch.Description
+}
+
+func (ch *HealthCheck) doChecks(parallel bool, result *HealthResult) {
```

Comment:
you don't need the `bool` param since the value is already available as `ch.Parallel`